services/api.ts

```diff
@@ -26,9 +28,16 @@ export async function fetchAvlHead(): Promise<{
   data: LatestBlockInfo["avlHead"];
 }> {
   const response = await fetch(`${appConfig.bridgeApiBaseUrl}/avl/head`);
-
+  //TODO: Change below as response.json() does not contain endTimestamp.
   const avlHead: LatestBlockInfo["avlHead"] = await response.json();
-  return { data: avlHead };
+  const api = await initialize(substrateConfig.endpoint);
+  const blockHash = await api.rpc.chain.getBlockHash(avlHead.data.end);
+  const block = await api.rpc.chain.getBlock(blockHash);
+  //Get first extrinsic which is timestamp.set, and extract the arg with timestamp.
+  const timestamp = parseInt(block.block.extrinsics[0].args[0].toJSON() as string);
```

Comment:
ah neat, so we take the timestamp from here. But we're going to compare it against Date.now() so clock skew could screw up the calculation, right?

I think it's an acceptable issue and ok to commit, though an idea for the future to make it more robust is that we could also try to calculate skew here by comparing Date.now() against the last block, which should give us accuracy up to the 20s blocktime or so.